.github/workflows/ci.yml

```diff
@@ -1,110 +1,205 @@
-        uses: olafurpg/setup-scala@v14
-        with:
-          java-version: adopt@1.11
-      - name: Cache scala dependencies
-        uses: coursier/cache-action@v6
-      - name: Lint code
-        run: sbt check
-
-  website:
-    runs-on: ubuntu-20.04
-    timeout-minutes: 60
+    - name: Git Checkout
+      uses: actions/checkout@v3.3.0
+      with:
+        fetch-depth: '0'
+    - name: Setup Scala
+      uses: actions/setup-java@v3.10.0
+      with:
+        distribution: temurin
+        java-version: '17'
+        check-latest: true
+    - name: Check if the README file is up to date
+      run: sbt  docs/checkReadme
+    - name: Check if the site workflow is up to date
+      run: sbt  checkGithubWorkflow
+    - name: Check artifacts build process
+      run: sbt  +publishLocal
+    - name: Check website build process
+      run: sbt docs/clean; sbt  docs/buildWebsite
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    continue-on-error: false
     steps:
-      - name: Checkout current branch
-        uses: actions/checkout@v3.2.0
-      - name: Setup Scala and Java
-        uses: olafurpg/setup-scala@v14
-        with:
-          java-version: adopt@1.11
-      - name: Cache scala dependencies
-        uses: coursier/cache-action@v6
-      - name: Check Document Generation
-        run: sbt docs/compileDocs
-
+    - name: Git Checkout
+      uses: actions/checkout@v3.3.0
+      with:
+        fetch-depth: '0'
+    - name: Setup Scala
+      uses: actions/setup-java@v3.10.0
+      with:
+        distribution: temurin
+        java-version: '17'
+        check-latest: true
+    - name: Lint
+      run: sbt lint
   test:
-    runs-on: ubuntu-20.04
-    timeout-minutes: 30
+    name: Test
+    runs-on: ubuntu-latest
+    continue-on-error: false
     strategy:
       fail-fast: false
       matrix:
-        java: [ 'adopt@1.8', 'adopt@1.11' ]
-        scala: [ '2.11.12', '2.12.17', '2.13.10', '3.2.2' ]
-        platform: [ 'JVM', 'JS' ]
+        java:
+        - '8'
+        - '11'
+        - '17'
+        scala:
+        - 2.12.17
+        - 2.13.10
+        - 3.2.2
     steps:
-      - name: Checkout current branch
-        uses: actions/checkout@v3.2.0
-        with:
-          fetch-depth: 0
-      - name: Setup Scala and Java
-        uses: olafurpg/setup-scala@v14
-        with:
-          java-version: ${{ matrix.java }}
-      - name: Cache scala dependencies
-        uses: coursier/cache-action@v6
-      - name: Mima Checks
-        if: ${{ !startsWith(matrix.scala, '3.') && endsWith(matrix.java, '.11') }}
-        run: ./sbt ++${{ matrix.scala }}! mimaChecks
-      - name: Run tests
-        if: ${{ endsWith(matrix.java, '.11') }}
-        run: sbt ++${{ matrix.scala }}! test${{ matrix.platform }}
-      - name: Run tests JVM8
-        if: ${{ endsWith(matrix.java, '.8') && matrix.platform == 'JVM' }}
-        run: sbt ++${{ matrix.scala }}! testJVM8
-      - name: Compile additional subprojects
```

Comment:
it is look like that this is missing in new version, if I see properly ....

Reply:
There is `+publishLocal` on line 39, but it doesn't compile against different java versions. If it is insufficient, I will add CI customization to the `zio-sbt-ci` plugin.

Reply:
@justcoon The question is when we are testing main components against various java versions, why do we need to test the compilation phase of their example projects?

Reply:
to check if they are compilable , as examples are in custom modules

Reply:
additional tests were incorporated.